src/control/cmd/dmg/pool.go

```diff
@@ -68,11 +70,17 @@ type PoolCreateCmd struct {
 
 // Execute is run when PoolCreateCmd subcommand is activated
 func (cmd *PoolCreateCmd) Execute(args []string) error {
-	if cmd.Size != "" && (cmd.ScmSize != "" || cmd.NVMeSize != "") {
-		return errIncompatFlags("size", "scm-size", "nvme-size")
+	var errIncompatFlagsNb int
+	for _, value := range [3]bool{cmd.Size != "", cmd.ScmSize != "" || cmd.NVMeSize != "", cmd.All} {
+		if value {
+			errIncompatFlagsNb++
+		}
+	}
+	if errIncompatFlagsNb > 1 {
+		return errIncompatFlags("size", "scm-size", "nvme-size", "all")
 	}
-	if cmd.Size == "" && cmd.ScmSize == "" {
-		return errors.New("either --size or --scm-size must be supplied")
+	if errIncompatFlagsNb < 1 {
+		return errors.New("either --size or --scm-size or --all must be supplied")
```

Comment:
You could also switch on the number of flags seen; might be slightly easier to read.

Reply:
Following tasks will be done:
- [ ] Refactoring sequential `i` with `switch`